capstone-ui/src/components/CreateGroup/CreateGroup.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from "react";
+import ProfileHeader from "../ProfileHeader/ProfileHeader";
+import "./CreateGroup.css";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import { logout, getGenres } from "../../spotify";
+import { formatDate } from "../../utils";
+import Switch from "react-switch";
+import Select from "react-select";
+import { catchErrors } from "../../utils";
+
+import logo from "../../logo.svg";
+
+export default function CreateGroup({
+  username,
+  token,
+  profile,
+  appProfile,
+  isPreferencesView,
+  tab,
+  setTab,
+  isFriendProfileView,
+}) {
+  const [groupName, setGroupName] = useState("");
+  const [description, setDescription] = useState("");
+  const [isPrivate, setIsPrivate] = useState(false);
+  const [genreOptions, setGenreOptions] = useState(null);
+  const [memberOptions, setMemberOptions] = useState(null);
+  const [selectedGenres, setSelectedGenres] = useState([]);
+  const [selectedMembers, setSelectedMembers] = useState([]);
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const allGenres = await getGenres();
+      let genreResults = allGenres.data.genres.map((element) => {
+        return { value: element, label: element };
+      });
+      setGenreOptions(genreResults);
+
+      const memberResponse = await axios.get(
+        `http://localhost:3001/user/followers/${username}`
+      );
+
+      let memberResults = memberResponse.data.map((element) => {
+        return { value: element, label: element };
+      });
+      setMemberOptions(memberResults);
+    };
+
+    catchErrors(fetchData());
+  }, []);
+
+  function createGroup() {
+    let postRequest = {
+      username: username,
+      groupName: groupName,
+      description: description,
+      isPrivate: isPrivate,
+      genres: selectedGenres,
+      isAdmin: isPrivate,
+    };
+    axios
+      .post("http://localhost:3001/user/group", postRequest)
+      .then(function (response) {
+        alert(`Success! Group ${groupName} was successfully created!`);
+      })
+      .catch((error) => {
+        alert(`Error! ${error.message}`);
+        return false;
+      });
+
+    setGroupName("");
+    setDescription("");
+    setIsPrivate(false);
+    setSelectedGenres([]);
+
+    for (let i = 0; i < selectedMembers.length; i++) {
+      let inviteRequest = {
+        username: selectedMembers[i].value,
+        groupName: groupName,
+      };
+      axios.post("http://localhost:3001/user/invite", inviteRequest).then();
+    }
+
+    setSelectedMembers([]);
+  }
+
+  function handleSwitch(e) {
+    setIsPrivate(e);
+  }
+
+  async function handleGenreChange(e) {
+    setSelectedGenres(e);
+  }
+
+  async function handleMemberChange(e) {
+    setSelectedMembers(e);
+  }
```

Comment:
variable names for parameters passed in should be specific. What does e represent?